classes/submission/Submission.inc.php

```diff
@@ -248,6 +248,50 @@ function setHideAuthor($hideAuthor) {
 	}
 
 	/**
+	 * Set comments to editor.
+	 * @param $commentsToEditor string
+	 */
+	function setCommentsToEditor($commentsToEditor) {
+		$this->setData('commentsToEditor', $commentsToEditor);
+	}
+
+	/**
+	 * Set submission revision.
+	 * @param $submissionRevision int
+	 */
+	function setSubmissionRevision($submissionRevision) {
+		$this->setData('submissionRevision', $submissionRevision);
+	}
+
+	/**
+	 * Return submission revision.
+	 * @return int
+	 */
+	function getSubmissionRevision() {
+		return $this->getData('submissionRevision');
+	}
+
+	/**
+	 * Get the current revision id for a submission
+	 * @param $contextId int
+	 * @return int
+	 */
+	function getCurrentVersionId($contextId = null) {
```

Comment:
contextId parameter is not needed, submission object already has it -- always belong to a context